lightbulb/commands/options.py

```diff
@@ -223,18 +227,27 @@ class Option(t.Generic[T, D]):
         :meth:`~attachment`
     """
 
-    __slots__ = ("_data", "_unbound_default")
+    __slots__ = ("_data", "_unbound_default", "_converter")
 
-    def __init__(self, data: OptionData[D], default_when_not_bound: T) -> None:
+    def __init__(self, data: OptionData[D], default_when_not_bound: T, converter: t.Callable[[D], types.MaybeAwaitable[T]] | None = None) -> None:
         self._data = data
         self._unbound_default = default_when_not_bound
+        self._converter = converter
 
     def __get__(self, instance: commands.CommandBase | None, owner: type[commands.CommandBase]) -> T | D:
         if instance is None or getattr(instance, "_current_context", None) is None:
             return self._unbound_default
 
-        return instance._resolve_option(self)
+        if not self._data._localized_name in instance._resolved_option_cache.keys():
+            raise RuntimeError(f"Tried to access option {self._data._localized_name} before resolving options.")
 
+        return t.cast("T", instance._resolved_option_cache[self._data._localized_name])
+
+    async def _convert(self, value: D) -> T:
+        try:
+            return t.cast("T", await maybe_await(di.with_di(self._converter)(value)))
+        except Exception as e:
+            raise ConversionFailedException(f"Failed to convert {value} for option {self._data._localized_name}") from e
```

Comment:
This isn't super useful I don't think, the developer has no information from which to generate an error response for the user. Maybe think about what should be included - param name, value, etc